engine/src/main/scala/cromwell/webservice/routes/CromwellApiService.scala

```diff
@@ -271,6 +248,50 @@ object CromwellApiService {
     }
   }
 
+  /**
+    * Sends a request to abort the workflow. Provides configurable success & error handlers to allow
+    * for different API endpoints to provide different effects in the appropriate situations, e.g. HTTP codes
+    * and error messages
+    */
+  def abortWorkflow(possibleWorkflowId: String,
+                    workflowStoreActor: ActorRef,
+                    workflowManagerActor: ActorRef,
+                    successHandler: WorkflowId => Route = standardAbortSuccessHandler,
+                    errorHandler: PartialFunction[Throwable, Route] = standardAbortErrorHandler)
+                   (implicit timeout: Timeout): Route = {
+    handleExceptions(ExceptionHandler(errorHandler)) {
+      Try(WorkflowId.fromString(possibleWorkflowId)) match {
+        case Success(workflowId) =>
+          val response = workflowStoreActor.ask(WorkflowStoreActor.AbortWorkflowCommand(workflowId)).mapTo[AbortResponse]
+          onComplete(response) {
+            case Success(x: SuccesfulAbortResponse) => successHandler(x.workflowId)
+            case Success(x: WorkflowAbortFailureResponse) => throw x.failure
+            case Failure(e) => throw e
+          }
+        case Failure(_) => throw InvalidWorkflowException(possibleWorkflowId)
+      }
+    }
+  }
+
+  /**
+    * The abort success handler for typical cases, i.e. cromwell's API.
+    */
+  private def standardAbortSuccessHandler(workflowId: WorkflowId): Route = {
+    complete(ToResponseMarshallable(WorkflowAbortResponse(workflowId.toString, WorkflowAborting.toString)))
+  }
+
+  /**
+    * The abort error handler for typical cases, i.e. cromwell's API
+    */
+  private def standardAbortErrorHandler: PartialFunction[Throwable, Route] = {
+    case e: InvalidWorkflowException => e.failRequest(StatusCodes.BadRequest)
+    case e: IllegalStateException => e.errorRequest(StatusCodes.Forbidden)
```

Comment:
I need to either bring back the comment from the old version, or remove this line if that comment is true (the comment says it's unreachable)

Reply:
done